pkg/trace/stats/concentrator.go

```diff
@@ -215,6 +228,59 @@ func (c *Concentrator) Add(t Input) {
 	c.mu.Unlock()
 }
 
+// ProcessOTLPTraces applies APM stats calculation on the otlp traces in the concentrator.
+// This function is NOT called in Concentrator.Run(), so you need to manually call it to get APM stats on the given OTLP traces.
+func (c *Concentrator) ProcessOTLPTraces(traces ptrace.Traces, additionalCtagKeys ...string) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	includeCID := shouldIncludeCIDDim(c.conf)
+	spanByID, resByID, scopeByID := traceutil.IndexOTelSpans(traces)
+	topLevelByKind := c.conf.HasFeature("enable_otlp_compute_top_level_by_span_kind")
+	topLevelSpans := traceutil.GetTopLevelOTelSpans(spanByID, resByID, topLevelByKind)
+	ignoreResNames := make(map[string]struct{})
+	for _, resName := range c.conf.Ignore["resource"] {
+		ignoreResNames[resName] = struct{}{}
+	}
+	for spanID, span := range spanByID {
+		res := resByID[spanID]
+		resName := traceutil.GetOTelResource(span, res)
+		if _, exists := ignoreResNames[resName]; exists {
+			continue
+		}
+		env := traceutil.GetOTelAttrValInResAndSpanAttrs(span, res, true, semconv.AttributeDeploymentEnvironment)
+		if env == "" {
+			env = c.agentEnv
+		}
+		aggKey := PayloadAggregationKey{
+			Env:      env,
+			Hostname: traceutil.GetOTelHostname(span, res, c.conf.OTLPReceiver.AttributesTranslator, c.conf.Hostname),
+			Version:  traceutil.GetOTelAttrValInResAndSpanAttrs(span, res, true, semconv.AttributeServiceVersion),
+		}
+		if includeCID {
+			if cid := traceutil.GetOTelAttrValInResAndSpanAttrs(span, res, true, semconv.AttributeContainerID, semconv.AttributeK8SPodUID); cid != "" {
+				aggKey.ContainerID = cid
+				if gitCommitSha, imageTag, err := version.GetVersionDataFromContainerTags(cid, c.conf); err == nil {
+					aggKey.GitCommitSha = gitCommitSha
+					aggKey.ImageTag = imageTag
+				}
+				ctags := traceutil.GetOTelContainerTags(res.Attributes(), additionalCtagKeys...)
```

Comment:
This is breaking change from users perspective as it will increase egress cost. We should send ctags for configured attributes , do not include the fixed set.

Reply:
I suspect if this is true. Today OTel spans need to go through trace/api/otlp.go then get APM stats. In  trace/api/otlp.go container tags are always appended: https://github.com/DataDog/datadog-agent/blob/ffb7c2ff293af5e2d2634e886977996a1d952fe9/pkg/trace/api/otlp.go#L297

It seems including the fixed set is backwards-compatible.

Reply:
Discussed offline, we will not include the default set of ctags.